// slack/index.js
var Promise = require('es6-promise').Promise;
var request = require('superagent');
var baseurl = require('baseurl').baseUrl;
var channels_join_url = "https://slack.com/api/channels.join";
var channels_history_url = "https://slack.com/api/channels.history";
var channel_message_url = "https://slack.com/api/chat.postMessage";

function slackRequest(opts) {
  var url = opts.url;
  var callback = opts.callback;

  if (!url) {
    throw new Error("requires a url");
  }
  if (typeof callback != "function") {
    throw new Error('requires a callback');
  }
  return request.get(url).end(callback);
}

module.exports.joinChannel = function joinChannel(name, callback) {
  return slackRequest({
    url: channels_join_url + "?token=" + this.props.token + "&name=" + name,
    callback: callback
  });
};

module.exports.channelHistory = function channelHistory(name, callback) {
  return slackRequest({
    url: channels_history_url + "?token=" + this.props.token + "&channel=" + name + "&count=10",
    callback: callback
  });
};

module.exports.messageChannel =

module.exports.parseOutgoing = function parseOutGoingMessage(message) {
  return message
};


module.exports.parseIncoming = function parseIncomingMessage(message) {
  if (message.subtype === "channel_leave" || message.subtype === "channel_join") {
  } else {
    return message
  }
};
